src/libpmem2/persist.c

```diff
@@ -321,8 +322,19 @@ pmem2_deep_flush_byte(struct pmem2_map *map, void *ptr, size_t size)
 {
 	LOG(3, "map %p ptr %p size %zu", map, ptr, size);
 
-	pmem2_persist_cpu_cache(ptr, size);
-	int ret = pmem2_deep_flush_dax(map);
+	/* XXX: read from map when available */
+	enum pmem2_file_type type;
+	int ret = pmem2_get_type_from_stat(&map->src_fd_st, &type);
+	if (ret) {
+		LOG(1, "cannot determine type of file from stat %p",
+				&map->src_fd_st);
+		return ret;
+	}
+
+	if (type == PMEM2_FTYPE_DEVDAX)
+		pmem2_persist_cpu_cache(ptr, size);
```

Comment:
I think we should always call flush, (in case of deep_flush is not supported) as deep flush should be as deep as possible. It shouldn't cost too much, as we are doing msync anyway. 

@pbalcer what do you think?

Reply:
Well, this operation returns error if it's not supported. If we were to ignore this error at this level, then sure. But I don't think failing operations should attempt to do anything else, this IMHO would be misleading.

And yes, this isn't very ergonomic...